frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/upsert-feature-flag-modal/upsert-feature-flag-modal.component.html

```diff
@@ -9,15 +9,27 @@
   <form [formGroup]="featureFlagForm" class="form-standard dense-3">
     <mat-form-field appearance="outline">
       <mat-label class="ft-14-400">Name</mat-label>
-      <input matInput formControlName="name" placeholder="e.g., My feature flag" class="ft-14-400"/>
+      <input
+        matInput
+        formControlName="name"
+        placeholder="e.g., My feature flag"
+        class="ft-14-400"
+        [disabled]="featureFlagForm.get('name')?.disabled"
```

Comment:
Do you need this line? As you have already disabled it from the component.

Reply:
The better way would be to create a function in the component 
```
isDisabled() {
      return this.config.params.action === UPSERT_FEATURE_FLAG_ACTION.EDIT &&
      this.config.params.sourceFlag?.status === FEATURE_FLAG_STATUS.ENABLED
}
```
and calling this function from the HTML